ts/features/bonus/cgn/screens/merchants/CgnMerchantsListScreen.tsx

```diff
@@ -96,59 +99,52 @@ const CgnMerchantsListScreen: React.FunctionComponent<Props> = (
     requestOnlineMerchants();
   }, [requestOfflineMerchants, requestOnlineMerchants]);
 
-  React.useEffect(initLoadingLists, [initLoadingLists]);
+  useFocusEffect(initLoadingLists);
 
   const onItemPress = (id: Merchant["id"]) => {
     props.navigateToMerchantDetail(id);
     Keyboard.dismiss();
   };
 
   return (
-    <BaseScreenComponent
-      goBack
-      headerTitle={I18n.t("bonus.cgn.merchantsList.navigationTitle")}
-      contextualHelp={emptyContextualHelp}
-    >
-      <SafeAreaView style={IOStyles.flex}>
-        {isReady(props.onlineMerchants) || isReady(props.offlineMerchants) ? (
-          <>
-            <View style={IOStyles.horizontalContentPadding}>
-              <H1>{I18n.t("bonus.cgn.merchantsList.screenTitle")}</H1>
-              <View style={{ height: 50 }}>
-                <LabelledItem
-                  icon="search"
-                  iconPosition={"right"}
-                  inputProps={{
-                    value: searchValue,
-                    autoFocus: false,
-                    onChangeText: setSearchValue,
-                    placeholder: I18n.t("global.buttons.search")
-                  }}
-                />
-              </View>
-            </View>
-            <CgnMerchantsListView
-              refreshing={
-                isLoading(props.onlineMerchants) ||
-                isLoading(props.offlineMerchants)
-              }
-              onRefresh={initLoadingLists}
-              merchantList={merchantList}
-              onItemPress={onItemPress}
+    <SafeAreaView style={IOStyles.flex}>
+      {isReady(props.onlineMerchants) || isReady(props.offlineMerchants) ? (
+        <>
+          <ContentWrapper>
+            <ListItemHeader
+              label={I18n.t("bonus.cgn.merchantsList.merchantsAll")}
+            />
+            <TextInput
+              accessibilityLabel={I18n.t("global.buttons.search")}
+              icon="search"
+              value={searchValue}
+              onChangeText={setSearchValue}
+              placeholder={I18n.t("global.buttons.search")}
+              autoFocus={false}
             />
-          </>
-        ) : (
-          <LoadingErrorComponent
-            isLoading={
-              isLoading(props.offlineMerchants) ||
-              isLoading(props.onlineMerchants)
+            <VSpacer />
+          </ContentWrapper>
```

Comment:
In the meantime the new filter screen will become available in a second iteration, what do you think to extract the search text input outside the remote condition? This is because if a user tries the pull-to-refresh, it happens that also the search input disappears for a while causing a lack of UX.

|Before|After|
|-|-|
|<video src="https://github.com/pagopa/io-app/assets/34343582/13ac936a-261e-4299-8466-7885c58ccba9" />|<video src="https://github.com/pagopa/io-app/assets/34343582/a0f20162-e999-47a6-a251-15cdd6cd1f18" />

Reply:
both addressed in [96c039b](https://github.com/pagopa/io-app/pull/5668/commits/96c039bbcaa1f3eb454e34fcde5864353615677f) and [1868144](https://github.com/pagopa/io-app/pull/5668/commits/18681443cd8a66e8cc7197231191f9e8208584a8)